Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,639 @@
+        self.map_canvas.setSelectionColor(QColor(Qt.cyan))
+        self.pan_tool = QgsMapToolPan(self.map_canvas)
+        self.map_canvas.setMapTool(self.pan_tool)
+
+        self.current_diff = None
+        self.diff_layers = []
+        self.filter_model = None
+        self.layer_list.currentRowChanged.connect(self.diff_layer_changed)
+
+        self.icons = {
+            "added": "plus.svg",
+            "removed": "trash.svg",
+            "updated": "pencil.svg",
+            "renamed": "pencil.svg",
+            "table": "table.svg",
+        }
+        self.model_detail = QStandardItemModel()
+        self.model_detail.setHorizontalHeaderLabels(["Details"])
+
+        self.details_treeview.setEditTriggers(QAbstractItemView.NoEditTriggers)
+        self.details_treeview.setModel(self.model_detail)
+
+        self.versionModel.current_version = self.mp.version()
+
+    def exec(self):
+
+        try:
+            ws_id = self.mp.workspace_id()
+        except ClientError as e:
+            QMessageBox.warning(None, "Client Error", str(e))
+            return
+
+        # check if user has permissions
+        try:
+            usage = self.mc.workspace_usage(ws_id)
+            if not usage["view_history"]["allowed"]:
+                QMessageBox.warning(
+                    None, "Upgrade required", "To view the project history, please upgrade your subscription plan."
+                )
+                return
+        except ClientError:
+            # Some versions e.g CE, EE edition doesn't have
+            pass
+        super().exec()
+
+    def closeEvent(self, event):
+        self.save_splitters_state()
+        QDialog.closeEvent(self, event)
+
+    def save_splitters_state(self):
+        settings = QSettings()
+        settings.setValue("Mergin/VersionViewerSplitterSize", self.splitter_map_table.saveState())
+        settings.setValue("Mergin/VersionViewerSplitterVericalSize", self.splitter_vertical.saveState())
+
+    def set_splitters_state(self):
+        settings = QSettings()
+        state_vertical = settings.value("Mergin/VersionViewerSplitterVericalSize")
+        if state_vertical:
+            self.splitter_vertical.restoreState(state_vertical)
+        else:
+            self.splitter_vertical.setSizes([120, 200, 40])
+
+        do_calc_height = True
+        state = settings.value("Mergin/VersionViewerSplitterSize")
+        if state:
+            self.splitter_map_table.restoreState(state)
+
+            if self.splitter_map_table.sizes()[0] != 0:
+                do_calc_height = False
+
+        if do_calc_height:
+            height = max([self.map_canvas.minimumSizeHint().height(), self.attribute_table.minimumSizeHint().height()])
+            self.splitter_map_table.setSizes([height, height])
+
+    def fetch_from_server(self):
+
+        if self.fetcher and self.fetcher.isRunning():
+            # Only fetching when previous is finshed
+            return
+        else:
+            self.fetcher.start()
```

Comment:
Is there any visual indication that we are fetching?

Reply:
Not at the moment, I think the cleanest way to do so would be to add a temporary items at the end of the version table Saying "Fetching more versions ⏳"